allofplos/transformations.py

```diff
@@ -178,9 +179,8 @@ def doi_to_path(doi, directory=None):
         article_file = os.path.join(directory, "plos.correction." + doi.split('/')[-1] + SUFFIX_LOWER)
     elif validate_doi(doi):
         article_file = os.path.join(directory, doi.lstrip(PREFIX) + SUFFIX_LOWER)
-    # NOTE: The following check is weird, a DOI should never validate as a file name.
-    elif validate_filename(doi):
-        article_file = doi
+    else:
```

Comment:
I really like the way you simplified the check with validate doi in `filename_to_doi`, could you use that method here as well? Checking for validation first that way you don't need to keep adding it as part of your conditions?